File: java-junit5/src/test/java/org/codecop/Session2b_WordCounterTest.java
package org.codecop;

import static org.junit.jupiter.api.Assertions.assertAll;
import static org.junit.jupiter.api.Assertions.assertEquals;
import static org.junit.jupiter.api.Assertions.assertIterableEquals;
import static org.junit.jupiter.api.Assertions.assertNotEquals;
import static org.junit.jupiter.api.Assertions.assertTimeout;

import java.time.Duration;
import java.util.Arrays;

import org.junit.jupiter.api.Test;

/**
 * Session 2b: WordCounterTest - Assertions new in JUnit 5. <br />
 * @see "https://junit.org/junit5/docs/current/user-guide/#writing-tests-assertions"
 */
class Session2b_WordCounterTest {

    // TODO Add the proper assertions to complete the tests. 
    // The test name explains what needs to be asserted. 

    @Test
    void shouldCountGreenTwice() {
        WordCounter counter = new WordCounter("green bar green hat");
        // TODO Check that 2, counter.countOf("green").
    }

    @Test
    void shouldNotCountCapitalizedWord() {
        WordCounter counter = new WordCounter("green bar green hat");
        // TODO Check that 1, counter.countOf("HAT") is not equal.
    }

    @Test
    void shouldFindUniqueWordAsSequence() {
        WordCounter counter = new WordCounter("green green");
        // TODO Check that Arrays.asList("green"), counter.uniqueWordsAsSequence() is iterable equal.
    }
    
    @Test
    void shouldReportSummaryOfWord() {
        WordCounter counter = new WordCounter("green green");
        WordCounter.Summary summary = counter.summaryOf("green");
        // TODO All these attributes of summary should be asserted at once.
                // TODO Check that "green", summary.word. //
                // TODO Check that 2, summary.count. //
    }

    @Test
    void shouldCountNumberOfWorldsIn100Millis() {
        WordCounter counter = new WordCounter("green bar green hat");
        // TODO Check that Duration.ofMillis(200), counter::numberOfWords has a timeout.
    }
    
}

File: csharp-xunit/WordCountTests/Session2b_WordCounterTest.cs
using Org.Codecop.WordCount;
using Xunit;

namespace Org.Codecop.WordCount.Tests
{
    /// <summary>Session 2b: WordCounterTest - More assertions.</summary>
    /// <seealso>"https://xunit.github.io/docs/comparisons#assertions"</seealso>
    public class Session2b_WordCounterTest
    {

        // TODO Add the proper assertions to complete the tests.
        // The test name explains what needs to be asserted.

        [Fact]
        public void ShouldCountGreenTwice()
        {
            var counter = new WordCounter("green bar green hat");
            // TODO Check that 2 and counter.CountOf("green") are equal.
        }

        [Fact]
        public void ShouldNotCountCapitalizedWord()
        {
            var counter = new WordCounter("green bar green hat");
            // TODO Check that 1 and counter.CountOf("HAT") are not equal.
        }

        [Fact]
        public void ShouldFindUniqueWordInUniqueWordSequence()
        {
            var counter = new WordCounter("green green");
            // TODO Check that "green" is contained in counter.UniqueWordsAsSequence().
        }

        [Fact]
        public void ShouldReturnRatioOfWordsAccuracyInRange()
        {
            var counter = new WordCounter("green bar green red green");
            // TODO Check that counter.RatioOf("bar") is in range 0.19, 0.21.
        }

        [Fact]
        public void ShouldReportSummaryOfWord()
        {
            var counter = new WordCounter("green green");
            var summary = counter.SummaryOf("green");
            // TODO Check that "green" and summary.word are equal.
            // TODO Check that 2 and summary.count are equal.
        }
    }
}
